### DISCUSSION TITLE
Adding More Cities To Auto Complete

### DISCUSSION THREAD

**Discussion by meseret3** [Imperial-lord/mapbox-flutter#4] [Q&A]
Hi hope you are all doing okay I want to add more cities to the autocomplete option of mapbox how do I do that I tried adding a tile but that does not work any help is greatly appreciated, thank you in advance

**Comment by Imperial-lord** [comment]
That is really easy, when you make the API request, your Request URL looks something like this - 
`https://api.mapbox.com/geocoding/v5/mapbox.places/Hi%20there.json?limit=10&types=place%2Cpostcode%2Caddress&access_token=YOUR_MAPBOX_ACCESS_TOKEN`

Here limit = 10 means you are showing 10 cities in the autocomplete (or search). You can vary this as per your need.

**Reply by meseret3** [reply]
Yeah but some cities aren't registered in mapbox so increasing the search limit won't help, is there any way I could add more cities.

**Reply by meseret3** [reply]
Is there a way to use Google maps api to get directions and pass the coordinates to mapbox navigation sdk

**Reply by Imperial-lord** [reply]
Alright, so there are 2 ways to get around this difficulty. 
1. As I have mentioned in my video, Mapbox allows you to make use of Mapbox Studio where you can make use of custom maps. Then you can add the cities you need, and then make use of that.
2. Yes as you have pointed out we could make use of a hybrid setup. You can make use of Google Maps Search API and then use reverse Geocoding to fetch the coordinates of those cities (or centre of the circle representing those cities). Then you can use the coordinates to create poly lines or whatever you want to use. There are 2 issues with this approach thought:
- Google Maps API key needs to be used so you will experience relatively higher costs
- Despite the fact that Map coordinates are mostly consistent, you might experience a slight difference in locations as you had expected on Google Maps if you plot those coordinates on Mapbox. This will be more prominent if you are using the coordinates to some remote locations.

**Reply by Imperial-lord** [reply]
I can write a small article on medium the how to do the second one if you need it? It'd be an interesting implementation!

**Reply by meseret3** [reply]
that would be really nice, that would be of great help!

**Reply by meseret3** [reply]
I tried adding the cities to a dataset in Mapbox but they don't come as an option in the autocomplete option.  I will be looking forward to reading that article

**Reply by Imperial-lord** [reply]
Sure, you can add yourself to my mailing list to get an email whenever I publish. Find me here - https://medium.com/@absatyaprakash01
